### PR TITLE
Openapi generator 7.0.1

### PR DESCRIPTION
**Pull request by altro3**
Update libraries.
Removed redundant dependencies.
Update commons-codec to remove vulnerability

### CODE REVIEW

**Review by melix — APPROVED** [review]

**Review by graemerocher — APPROVED** [review]

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -9,10 +9,11 @@ jspecify = "0.3.0"
 jdt-annotation = "2.2.700"
 android-annotation = "1.7.0"
 spotbugs-annotations = "4.7.3"
-openapi-generator = "6.6.0"
+openapi-generator = "7.0.0"
 swagger-parser = "1.0.67"
 swagger-parser-v3 = "2.1.16"
 javaparser = "3.25.5"
+commons-codec = "1.16.0"
```
Comment (sdelamo) [inline gradle/libs.versions.toml:16]: `commons-codec` is not necessary or used is it?

**Review by sdelamo — CHANGES_REQUESTED** [review]

File: gradle/libs.versions.toml (hunk 1 above)
Comment (altro3) [reply gradle/libs.versions.toml:16]: @sdelamo It is used by openapi-generator, not remembered directly or through swagger-parser. In any case, the old version has a vulnerability. This fix will help remove this vulnerability from micronaut-openapi

Comment (sdelamo) [reply gradle/libs.versions.toml:16]: do you know about this? @melix

Comment (altro3) [reply gradle/libs.versions.toml:16]: @melix ping

**Review by melix — APPROVED** [review]
Ideally `commons-codec` should be a constraint. However this will not work with Maven, so making it an `implementation` dependency makes sense.

**Review by sdelamo — APPROVED** [review]

**Comment by sdelamo** [comment]
fails because of https://github.com/micronaut-projects/micronaut-openapi/pull/1217

**Comment by altro3** [comment]
I already fixed it here

**Review by melix — APPROVED** [review]

**Pull request merged**